airflow/models/baseoperator.py

```diff
@@ -672,6 +709,21 @@ class derived from this one results in the creation of a task object,
         If set to `None` (default), the logger name will fall back to
         `airflow.task.operators.{class.__module__}.{class.__name__}` (e.g. SimpleHttpOperator will have
         *airflow.task.operators.airflow.providers.http.operators.http.SimpleHttpOperator* as logger).
+    :param allow_mixin: if True, when an operator is executed within another one a warning message will be
```

Comment:
After the discussion, I think having a warning (i.e. True) by default here is a safer approach. Seems that a number of people might not care currently that there is no templating and use the operators this way, I think warning now and possibly turning that into working feature in the future might be a better idea than disallowing it. Simply the number of people hit by this one might be quite big - we might not realize how many people are doing that already.

One more thing - I thin `allow_mixin` is likely not best name. Maybe `allow nested operators`  following the naming used by one of the users would be a good idea.

Reply:
done